chef/cookbooks/nova_dashboard/recipes/server.rb

```diff
@@ -17,6 +17,30 @@
 include_recipe "apache2::mod_wsgi"
 include_recipe "apache2::mod_rewrite"
 
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+# Multidomain can only be enabled if the API version of Keystone is
+# equal or bigger that v3.
+keystone_api_version = keystone_settings['api_version']
+keystone_internal_auth_url = keystone_settings['internal_auth_url']
+
+# TODO(aplanas): Remove this line when keystone API 'v3' works for
+# every OpenStack configuration file.
+if node["nova_dashboard"]["multidomain"] && keystone_api_version.to_f < 3
+  keystone_api_version = '3'
+  keystone_internal_auth_url = KeystoneHerper.versioned_service_URL(node,
```

Comment:
Really useful.  I will refactor `KeystoneHelper.versioned_service_URL` and test it in c16. Thanks!